### PR TITLE
Fix  undefined method `+' for nil:NilClass\n\nProfile: - when using profile dependencies and require_controls

### PR DESCRIPTION
**Pull request by Vasu1105**
Signed-off-by: Vasu1105 <vasundhara.jagdale@chef.io>

## Description

While using the dependencies if we add one more profile inside controls and use require_controls the InSpec scan breaks.

Example
```
inheritance
|-- controls
   |-- dev.rb
   |-- stage.rb
|-- inspec.yml
|-- profile_group_1
   |-- controls
      |-- c1.rb
   |-- inspec.yml
```
# dev.rb
```
require_controls 'group_1' do
  control 'c1'
end
```
# stage.rb
```
require_controls 'group_1' do
  control 'c1'
end
```
# inspec.yml
```
name: inheritance
version: 0.1.0
depends:
  - name: group_1
     path: ./profile_group_1
```
# c1.rb
```
control 'c1' do
  title 'hello'
  desc 'world'
  impact 0.5
  describe 'HELLO' do
    it { should cmp 'HELLO' }
  end
end
```

After running the below command 

```
inspec exec --no-create-lockfile
```

Following failure message is shown

```
[2021-04-27T15:11:06+05:30] ERROR: Failed to load profile inheritance: Failed to load source for controls/stage.rb: undefined method `+' for nil:NilClass

Profile:         InSpec Profile (inheritance)
Version:         0.1.0
Failure Message: Failed to load source for controls/stage.rb: undefined method `+' for nil:NilClass
Target:          local://

     No tests executed.
```
We can see the failing test here due to above error here https://buildkite.com/chef-oss/inspec-inspec-master-verify/builds/3010#b881f718-2a22-4cfd-965d-cf497306ced9/750-766

## Related Issue
This resolves small partt of #4318

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New content (non-breaking change)
- [ ] Breaking change (a content change which would break existing functionality or processes)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: lib/inspec/profile_context.rb [hunk 1]
```diff
@@ -91,7 +91,7 @@ def remove_rule(id)
     end
 
     def all_controls
-      ret = @rules.values
+      ret = @rules.values.reject(&:nil?)
```
Comment (Nik08) [inline lib/inspec/profile_context.rb]: Seems like we could have also used `@rules.values.compact` here in this case? Even using reject will not have any implications, it's the same. It's just `compact` appears more neat. Just my opinion. What are your thoughts?

**Review by Nik08 — COMMENTED** [review]
@Vasu1105 Is it also possible to add a test case for testing this profile dependency?

File: lib/inspec/profile_context.rb (hunk 1 above)
Comment (Vasu1105) [reply lib/inspec/profile_context.rb]: That can be done. Thanks @Nik08

**Review by clintoncwolfe — APPROVED** [review]

**Comment by Vasu1105** [comment]
@Nik08 I added the additional profile for that and already existing test case is capturing that.

**Comment by Nik08** [comment]
@Vasu1105 Oh because of the additional profile, I thought there will be some additional test cases. Thanks for the clarification :)

**Pull request merged**